src/main/java/com/linkedin/kafka/clients/utils/DefaultHeaderSerde.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+package com.linkedin.kafka.clients.utils;
+
+import java.nio.ByteBuffer;
+
+
+/**
+ * Utilities to implement header serializer and deserializer.
+ */
+public interface DefaultHeaderSerde {
```

Comment:
All these constants really have to do with how the header is encoded and not constants that deal with all possible encodings.  For example, you could encode the headers as UTF-16 which is Java's native encoding which means more overhead for ascii characters, but less copying on the consumer side.  Magic numbers are very specific to our encoding.   I don't see anything that belongs in HeaderUtils.